core/chat/persistence.go

```diff
@@ -61,6 +67,9 @@ func getChatHistory() []models.ChatMessage {
 	if err != nil {
 		log.Fatal(err)
 	}
+	if rows.Err() != nil {
```

Comment:
Is this this different than the `err`'s value? If so, might change it to be:

`if err := rows.Err(); err != nil {`

Reply:
It is different indeed: it will return the last error that occurred during iteration over the result set. It should be called after the `for rows.Next()` loop to ensure that we really handled all rows in the result.

Reply:
Nice catch!